File: web/tests/reference/greedy/jump-game.js
function canJump(nums) {
  let maxReach = 0;
  const last = nums.length - 1;
  for (let i = 0; i < nums.length; i++) {
    if (i > maxReach) return false;
    maxReach = Math.max(maxReach, i + nums[i]);
    if (maxReach >= last) return true;
  }
  return true;
}

File: web/tests/reference/greedy/jump-game.py
def can_jump(nums):
    max_reach = 0
    last = len(nums) - 1
    for i, n in enumerate(nums):
        if i > max_reach:
            return False
        max_reach = max(max_reach, i + n)
        if max_reach >= last:
            return True
    return True
